src/OrleansDockerUtils/DockerMembershipOracle.cs

```diff
@@ -0,0 +1,468 @@
+﻿using Microsoft.Orleans.Docker.Models;
+using Microsoft.Orleans.Docker.Utilities;
+using Orleans;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Orleans.Docker
+{
+    internal class DockerMembershipOracle : IMembershipOracle, IDockerStatusListener, IDisposable
+    {
+        /// <summary>
+        /// Status of this silo.
+        /// </summary>
+        public SiloStatus CurrentStatus => GetApproximateSiloStatus(SiloAddress);
+
+        /// <summary>
+        /// Address of this silo.
+        /// </summary>
+        public SiloAddress SiloAddress => localSiloDetails.SiloAddress;
+        
+        /// <summary>
+        /// Name of this silo.
+        /// </summary>
+        public string SiloName => localSiloDetails.Name;
+
+        private readonly ConcurrentDictionary<ISiloStatusListener, ISiloStatusListener> subscribers =
+            new ConcurrentDictionary<ISiloStatusListener, ISiloStatusListener>();
+        private readonly object updateLock = new object();
+        private readonly Dictionary<SiloAddress, SiloEntry> silos = new Dictionary<SiloAddress, SiloEntry>();
+        private readonly TimeSpan refreshPeriod = TimeSpan.FromSeconds(30);
+        private readonly ILocalSiloDetails localSiloDetails;
+        private readonly Logger log;
+        private readonly GlobalConfiguration globalConfig;
+        private readonly IDockerSiloResolver resolver;
```

Comment:
Just to be sure, which parts of this file differ from the Service Fabric oracle other than this type?